projects/medical/2d_image/dermoscopy/ddti/README.md

```diff
@@ -0,0 +1,155 @@
+
+| Dataset name                                                                   | Anatomical region | Task type    | Modality   | Num. Classes | Train/Val/Test Images | Train/Val/Test Labeled | Release Date | License                                                         |
+| ------------------------------------------------------------------------------ | ----------------- | ------------ | ---------- | ------------ | --------------------- | ---------------------- | ------------ | --------------------------------------------------------------- |
+| [ddti](https://drive.google.com/file/d/1wwlsEhwfSyvQsJBRjeDLhUjqZh8eaH2R/view) | other             | segmentation | dermoscopy | 2            | 831/-/278             | yes/-/yes              | 2020         | [CC-BY-NC 4.0](https://creativecommons.org/licenses/by-sa/4.0/) |
+
+|   Class Name   | Num. Train | Pct. Train | Num. Val | Pct. Val | Num. Test | Pct. Test |
+| :------------: | :--------: | :--------: | :------: | :------: | :-------: | :-------: |
+|   background   |    831     |   97.28    |    -     |    -     |    278    |   97.21   |
+| nodularThyroid |    779     |    2.72    |    -     |    -     |    261    |   2.79    |
+
+Note:
+
+- `Pct` means percentage of pixels in this category in all pixels.
+
+### Visualization
+
+![bac](https://raw.githubusercontent.com/uni-medical/medical-datasets-visualization/main/2d/semantic_seg/dermoscopy/ddti/ddti_dataset.png)
+
+## Usage
+
+### Prerequisites
+
+- Python v3.8
+- PyTorch v1.10.0
+- pillow (PIL) v9.3.0
+- scikit-learn (sklearn) v1.2.0
+- [MIM](https://github.com/open-mmlab/mim) v0.3.4
+- [MMCV](https://github.com/open-mmlab/mmcv) v2.0.0rc4
+- [MMEngine](https://github.com/open-mmlab/mmengine) v0.2.0 or higher
+- [MMSegmentation](https://github.com/open-mmlab/mmsegmentation) v1.0.0rc5
+
+All the commands below rely on the correct configuration of `PYTHONPATH`, which should point to the project's directory so that Python can locate the module files. In `ddti/` root directory, run the following line to add the current directory to `PYTHONPATH`:
+
+```shell
+export PYTHONPATH=`pwd`:$PYTHONPATH
+```
+
+### Dataset Preparing
+
+- Download dataset from [here](https://drive.google.com/file/d/1wwlsEhwfSyvQsJBRjeDLhUjqZh8eaH2R/view) and save it to the `data/` directory .
+- Decompress data to path `data/`. This will create a new folder named `data/ddti/`, which contains the original image data.
+- run script `python tools/prepare_dataset.py` to format data and change folder structure as below.
+- run script `python ../../tools/split_seg_dataset.py` to split dataset. For the Bacteria_detection dataset, as there is no test or validation dataset, we sample 20% samples from the whole dataset as the validation dataset and 80% samples for training data and make two filename lists `train.txt` and `val.txt`. As we set the random seed as the hard code, we eliminated the randomness, the dataset split actually can be reproducible.
```

Comment:
After step 2, the content structure is like this:
```shell
data/DDTI
├── 1_or_data
│   ├── category.csv
│   ├── image
│   ├── mask
│   └── readme.txt
├── 2_preprocessed_data
│   ├── readme.txt
│   ├── stage1
│   ├── stage2
│   └── train.csv
└── readme.txt
```
It seems that can not directly use the `split_seg_dataset.py` script.

Reply:
There is something wrong with our current dataset saved in ssd. I will check it later.